### ISSUE TITLE
Unable to remove Ammo module reference?

### ISSUE THREAD

**Issue by LeXXik** [playcanvas/editor#559]
One of the projects was working fine when I left it yesterday, but after returning to it today, it was freezing on start - the loading progress bar would hang indefinitely, freezing the launcher along with any other tab that has playcanvas domain opened.

The issue appeared to be in Ammo. There are no logs, it just immediately freezes as soon as it is opened. I went ahead and removed the module (by deleteing the Ammo folder from the Assets paenl), and wanted to add it anew, but it says I already have it and adding another one could introduce a duplicate.

I'm pretty sure I don't have another Ammo files (js/wasm) in the project. If I add a new version, what will happen? Will there be a sort of invalid duplicate reference together with a valid one? I tried to add a new one and then remove it again - I still have that error message, but at least it doesn't freeze.

https://user-images.githubusercontent.com/5677782/135448348-ba0ffcd1-386a-424b-804c-a677bbbde36f.mp4

**Comment by yaustar** [comment]
Are there any editor logs/errors?

**Comment by LeXXik** [comment]
No logs on Editor. I will DM you.

**Comment by yaustar** [comment]
Repro'ed in a new project: https://playcanvas.com/editor/scene/1241238

1. Create a new project
2. Create a script that starts with the word 'ammo'
3. Try to import Ammo module.
4. See error message 

![image](https://user-images.githubusercontent.com/16639049/135453693-55645582-fcf2-4c6f-8a92-1b5853098335.png)

This is because the code we have in the Editor checks for scripts that start with 'ammo' which includes any user generated scripts:

```
    // method if the module is included in the project
    editor.method('project:module:hasModule', function (wasmFilename) {
        var moduleAssets = editor.call('assets:find', function (item) {
            var name = item.get('name');
            var type = item.get('type');
            return name && name.indexOf(wasmFilename) >= 0 && (type === 'script' || type === 'wasm');
        });
        return moduleAssets.length > 0;
    });
```

**Comment by slimbuck** [comment]
Yep this is works-as-expected. Perhaps the message should be re-worded to indicate this is a warning not an error?

**Comment by yaustar** [comment]
Rewording it to have 'Please continue if you have other scripts or assets starting with 'ammo' that you know aren't the Ammo WASM module' would help I think

**Comment by yaustar** [comment]
Random idea, instead of checking for filenames, could we check against the module name of all the current WASM modules in the project? That would solve the name clash issue and would also work with Basis and any other future WAS modules we add: 

<img width="534" alt="image" src="https://user-images.githubusercontent.com/16639049/154278908-40641790-9d67-4c52-b34d-a15e763514e6.png">

**Comment by slimbuck** [comment]
> Random idea, instead of checking for filenames, could we check against the module name of all the current WASM modules in the project? That would solve the name clash issue and would also work with Basis and any other future WAS modules we add:
> 
> <img alt="image" width="534" src="https://user-images.githubusercontent.com/16639049/154278908-40641790-9d67-4c52-b34d-a15e763514e6.png">

Yes this is a great idea. Wish we'd done it this way originally.

**Comment by LeXXik** [comment]
Throwing it here, no clue if useful :) Those WASM components are only working with Emscripten generated WASM files. They then get assigned to global space window.Basis, like any other global variable a dev might occupy. WASM modules generated by other tools, like from Rust language or AssemblyScript, are instantiated by devs manually, since those can't be used in in this component module in most of the cases. Also, if Ammo is loaded later, then that component is not used as well.
